elastic/apm-server
# sampling: fix keep_unsampled processor

## Motivation/summary

Fix a panic which occurs when `apm-server.sampling.keep_unsampled: false` is configured.

## Checklist


I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

Run server with `apm-server.sampling.keep_unsampled: false`, and send a series of unsampled transactions. The server should not log any panics.

## Related issues

None.
